### ISSUE TITLE
Do you want to run a builder? 

### ISSUE THREAD

**Issue by come-maiz** [flashbots/mev-boost#145]
So, we heard you want to run a builder! Tell us more :)

The [builder API specification](https://ethereum.github.io/builder-specs/) is open and every validator can select their own builders. However, there are still open questions that we need to resolve with the humans in the ethereum ecosystem to get a permissionless and fully decentralized design. Following [the plan](https://github.com/flashbots/mev-boost/wiki/The-Plan-(tm)) towards proposer/builder separation we are [researching and prototyping](https://github.com/flashbots/mev-boost/wiki/Research) solutions for decentralization. Coming next is the addition of independent block builders to the Flashbots relay. You can find the initial [Relay API specification here](https://flashbots.notion.site/Relay-API-Spec-5fb0819366954962bc02e81cb33840f5), which outlines how builders can interact with the Flashbots Relay.

Tell us more about who you are, what is your role in the ecosystem, how will you profit from running a builder, how are you going to contribute to answer the [open questions](https://github.com/flashbots/mev-boost/wiki/Research#open-questions), in order to get us to the trustless proposer/builder separation. Leave a comment in this issue so we can talk about collaboration.

**Comment by Code0x2** [comment]
Hi,
I'm currently a searcher operating a few strats around the place, and interested in operating a builder. I believe that I can as a builder by constructing blocks with profitable txs created by my bots. However, I think that moving into an aggregated block system rather than the current aggregated relay would mean generally less profit for block producers and less inclusion for off-chain mev txs, as a majority of mev extraction txs would not be able to be seen by a builder, since there would be many and searchers either would have to run their own builder or go back to PGA. Additionally, if a searcher was not to run a builder, or for users interested in a private tx service they would have to build trusted relationships with certain builders, to reduce leaking of txs and other harmful activities, which would reduce inclusion

**Comment by jeromelaurens** [comment]
Hi the flashbot team, 
we are a small team, mainly doing searching.
We are currently considering running a builder, not sure yet.
We would be happy to exchange and may be contribute

**Comment by xcarlo** [comment]
bloXroute is planning to run its own builder and a relay

**Comment by xcarlo** [comment]
> > bloXroute is planning to run its own builder and a relay
> 
> great news ! Do you plan to accept external builders or only your own ?

Please send a message to `eyal@bloxroute.com` or discord `Eyal|bloXroute#0194` for more information.

**Comment by sourdzl** [comment]
we are interesting in implementing a relay, and for our initial version will include a naive builder internal to the relay server.  are there any details you can provide on helping guide our relay implementation?  we have seen the HTTP api interface for the relay, but that's it so far!

**Comment by block-builder-22** [comment]
> So, we heard you want to run a builder! Tell us more :)
> 
> The [builder API specification](https://ethereum.github.io/builder-specs/) is open and every validator can select their own builders. However, there are still open questions that we need to resolve with the humans in the ethereum ecosystem to get a permissionless and fully decentralized design. Following [the plan](https://github.com/flashbots/mev-boost/wiki/The-Plan-(tm)) towards proposer/builder separation we are [researching and prototyping](https://github.com/flashbots/mev-boost/wiki/Research) solutions for decentralization. Coming next is the addition of independent block builders to the Flashbots relay.
> 
> Tell us more about who you are, what is your role in the ecosystem, how will you profit from running a builder, how are you going to contribute to answer the [open questions](https://github.com/flashbots/mev-boost/wiki/Research#open-questions), in order to get us to the trustless proposer/builder separation. Leave a comment in this issue so we can talk about collaboration.



Hi all!

**Who are we & what is our role in the ecosystem:**
We are a ventured backed crypto trading firm that specializes in atomic intra-block and cross chain arbitrage as well as a plethora of other on-chain strategies like liquidations and exotics. We have invested a lot of resources into infrastructure to be able compete at a very high level and would like to use our experience in MEV as a searcher as well as some of our infrastructure e.g. custom high performance EVM implementations, to run a builder. 

**How will we profit from block building:**
We have developed a novel model for a block builder marketplace with different characteristics compared to flashbots sealed bid blockspace auction. 
We believe features like fair ordering, minimization of frontrunning/sandwiching and other unique features that we have in the works will be an important value add to the space and allow both users and searchers to have a more diverse set of choices when it comes to expressing a preference of how transactions/bundles will be handled for block inclusion. There are some hurdles that we will have to overcome at the beginning to bootstrap the marketplace and be competitive, so we would like to start prototyping and building as soon as possible (ideally on some of the testnets before the merge)


**How are we going to contribute to answer some of the open questions:**
Block construction is going to be a core focus for our business, hence we will dedicate full-time resources towards integrating with the builder spec etc and expect to be working closely with Flashbots/EF. What excites us about being early is that it will will allow us to contribute and help shape how block construction as an industry will evolve.

**TLDR:**
We believe that PBS (and MEV-Boost as an interim step) is one of the most exciting things happening in the space and are very keen to be part of it from the begging!

@elopio  @thegostep  @astarinmymind   Plz DM me in regards to how we can best get started. 

Discord handle: Builder#4439

PS: Anyone reading this who is as excited about block construction as we are and has the technical chops to build cool tech and is interested in joining a our team. DM's are open

**Comment by sambacha** [comment]
we are already running one, come join manifoldfinance.com

**Comment by bertmiller** [comment]
Hi all, raising a related issue with how we will implement spam protection at the Flashbots Relay for builders here: https://github.com/flashbots/mev-boost/issues/219

**Comment by ntop001** [comment]
> Hi, I'm currently a searcher operating a few strats around the place, and interested in operating a builder. I believe that I can as a builder by constructing blocks with profitable txs created by my bots. However, I think that moving into an aggregated block system rather than the current aggregated relay would mean generally less profit for block producers and less inclusion for off-chain mev txs, as a majority of mev extraction txs would not be able to be seen by a builder, since there would be many and searchers either would have to run their own builder or go back to PGA. Additionally, if a searcher was not to run a builder, or for users interested in a private tx service they would have to build trusted relationships with certain builders, to reduce leaking of txs and other harmful activities, which would reduce inclusion

I have the same concerns as you, big MEV team will be more competitive than small or single searcher if everyone can run a builder.

**Comment by come-maiz** [comment]
@ntop001, @sambacha, @block-builder-22, @sourdzl, @xcarlo, @jeromelaurens, @Code0x2, would you like to test your builders in Goerli sending blocks to the Flashbots relay?

We are ready for this next phase of testing, and would like to start with you before opening it to others.

**Comment by block-builder-22** [comment]
> @ntop001, @sambacha, @block-builder-22, @sourdzl, @xcarlo, @jeromelaurens, @Code0x2, would you like to test your builders in Goerli sending blocks to the Flashbots relay?
> 
> We are ready for this next phase of testing, and would like to start with you before opening it to others.

Great, we would be interested in testing the relay. How do we get the access details?

**Comment by sambacha** [comment]
> @ntop001, @sambacha, @block-builder-22, @sourdzl, @xcarlo, @jeromelaurens, @Code0x2, would you like to test your builders in Goerli sending blocks to the Flashbots relay?
> 
> We are ready for this next phase of testing, and would like to start with you before opening it to others.

put me in coach

**Comment by TennisBowling** [comment]
Hey, I'm a solo staker who has a couple validators (so some skin in the game :) ). I've also developed executionbackup, letting a 1:many CL to EL. I also have some more projects coming up ;)
as far as independent goes, I don't have any affiliation with anything. I wish to benefit ethereum and stakers.

**Comment by TennisBowling** [comment]
Oh and, I might be thinking of making a builder implem in c++

**Comment by metachris** [comment]
The [Flashbots Goerli relay](https://builder-relay-goerli.flashbots.net/) now allows block submissions by external builders 🎉 

See also [Relay API documentation - Block Builder API](https://flashbots.notion.site/Relay-API-Spec-5fb0819366954962bc02e81cb33840f5#854339c909a042d0bbca6e8f8069674e)

The endpoints are:

* `getValidators`: https://builder-relay-goerli.flashbots.net/relay/v1/builder/validators (GET)
* `submitBlock`: https://builder-relay-goerli.flashbots.net/relay/v1/builder/blocks (POST)

Rate-limit for submissions is currently 10 / minute, will allow more soon (the goal is to allow ~ 1 request per sec).

**Comment by sambacha** [comment]
> The [Flashbots Goerli relay](https://builder-relay-goerli.flashbots.net/) now allows block submissions by external builders 🎉
> 
> See also [Relay API documentation - Block Builder API](https://flashbots.notion.site/Relay-API-Spec-5fb0819366954962bc02e81cb33840f5#854339c909a042d0bbca6e8f8069674e)
> 
> The endpoints are:
> 
> * `getValidators`: https://builder-relay-goerli.flashbots.net/relay/v1/builder/validators (GET)
> * `submitBlock`: https://builder-relay-goerli.flashbots.net/relay/v1/builder/blocks (POST)
> 
> Rate-limit for submissions is currently 10 / minute, we'll allow more soon (the goal is to allow ~ 1 request per sec).

Thanks @thegostep was wondering if I was going to have to force the issue 💯😘

**Comment by metachris** [comment]
Increased the rate-limit to 30 blocks / minute.

**Comment by 0xhillybilly** [comment]
Hi!

0xbeefbabe is interested in running a builder. As the top fee spending MEV bot we will be able to provide competitive blocks.

We have experimented on the Goerli relay and will be ready to submit full blocks from day 1 post merge. Eagerly awaiting information from Flashbots on the availability and onboarding process of the relay API endpoints.

Can get in touch through hillyhilbert@gmail.com

**Comment by come-maiz** [comment]
Hello humans,

We have just opened a forum for builders self support, so we can get together, answer the remaining questions, and report about our experiences building blocks. Please join here:

https://collective.flashbots.net/c/builders/14

**Comment by hawkj167** [comment]
Hi guys!

Our team is thinking of running a builder. We have experience in the MEV searcher for two years. Last week we set up and maintained a block builder on the Goerli. After the merge, we are happy to take the democracy of MEV distribution a step further.

Feel free to contact us through jackson167smith@gmail.com.

**Comment by metachris** [comment]
In addition to our Goerli relay, **we've now also opened block submissions on our [Sepolia relay](https://builder-relay-sepolia.flashbots.net/)** (where our relay delivers about 1 in 5 blocks).

**Sepolia relay block submission endpoints:**

* `getValidators`: https://builder-relay-sepolia.flashbots.net/relay/v1/builder/validators (GET) - array of proposer preferences for registered validators with assigned duties in the current and next epoch
* `submitBlock`: https://builder-relay-sepolia.flashbots.net/relay/v1/builder/blocks (POST)

**Goerli relay block submission endpoints:**

* `getValidators`: https://builder-relay-goerli.flashbots.net/relay/v1/builder/validators (GET) - array of proposer preferences for registered validators with assigned duties in the current and next epoch
* `submitBlock`: https://builder-relay-goerli.flashbots.net/relay/v1/builder/blocks (POST)

**See also:**

* [Relay API documentation - Block Builder API](https://bit.ly/3BmGZ3T)


---

Note: Submissions are currently rate-limited to 60 requests / minute / IP address.

**Comment by metachris** [comment]
The Flashbots mainnet relay will also allow external builder submissions from the get-go (the merge), in the same way as our testnet relays.

**Comment by metachris** [comment]
**Mainnet relay block submission endpoints:**

- `getValidators`: [https://boost-relay.flashbots.net/relay/v1/builder/validators](https://boost-relay.flashbots.net/relay/v1/builder/validators) (GET) - array of proposer preferences for registered validators with assigned duties in the current and next epoch
- `submitBlock`: [https://boost-relay.flashbots.net/relay/v1/builder/blocks](https://boost-relay.flashbots.net/relay/v1/builder/blocks) (POST)

**Comment by metachris** [comment]
Closing this issue, further discussions about running a builder happen in https://collective.flashbots.net/c/builders/14
